src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -94,613 +71,275 @@ private static bool FirstCharInUInt32IsAscii(uint value)
+            return bufferLength;
 
-            pBuffer += SizeOfVector128;
+        Found512:
+            found512 = Vector512.GreaterThanOrEqual(found512, Vector512.Create((byte)0x80));
+            return ComputeIndex(found512) + (nuint)Unsafe.ByteOffset(ref pBuffer, ref pCurrent);
 
-        DoFinalUnalignedVectorRead:
-
-            if (((byte)bufferLength & MaskOfAllBitsInVector128) != 0)
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            static bool Test128(Vector128<byte> value)
             {
-                // Perform an unaligned read of the last vector.
-                // We need to adjust the pointer because we're re-reading data.
-
-                pBuffer += (bufferLength & MaskOfAllBitsInVector128) - SizeOfVector128;
-
                 if (Sse2.IsSupported)
                 {
-                    currentSseMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
-                    if (ContainsNonAsciiByte_Sse2(currentSseMask))
-                    {
-                        goto FoundNonAsciiDataInCurrentChunk;
-                    }
-
-                }
-                else if (AdvSimd.Arm64.IsSupported)
-                {
-                    currentAdvSimdIndex = (uint)GetIndexOfFirstNonAsciiByteInLane_AdvSimd(AdvSimd.LoadVector128(pBuffer), bitmask); // unaligned load
-                    if (ContainsNonAsciiByte_AdvSimd(currentAdvSimdIndex))
-                    {
-                        goto FoundNonAsciiDataInCurrentChunk;
-                    }
-
-                }
-                else
-                {
-                    throw new PlatformNotSupportedException();
+                    return value.ExtractMostSignificantBits() != 0;
                 }
 
-                pBuffer += SizeOfVector128;
+                return Vector128.GreaterThanOrEqualAny(value, Vector128.Create((byte)0x80));
             }
 
-        Finish:
-            return (nuint)pBuffer - (nuint)pOriginalBuffer; // and we're done!
-
-        FoundNonAsciiDataInInnerLoop:
-
-            // If the current (first) mask isn't the mask that contains non-ASCII data, then it must
-            // instead be the second mask. If so, skip the entire first mask and drain ASCII bytes
-            // from the second mask.
-
-            if (Sse2.IsSupported)
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            static bool Test256(Vector256<byte> value)
             {
-                if (!ContainsNonAsciiByte_Sse2(currentSseMask))
+                if (Avx.IsSupported)
                 {
-                    pBuffer += SizeOfVector128;
-                    currentSseMask = secondSseMask;
+                    return value.ExtractMostSignificantBits() != 0;
                 }
-            }
-            else if (AdvSimd.IsSupported)
-            {
-                if (!ContainsNonAsciiByte_AdvSimd(currentAdvSimdIndex))
+
+                if (Vector256.IsHardwareAccelerated)
```

Comment:
We shouldn't be using `V256` if it isn't hardware accelerated. Doing so can throttle the CPU, miss optimizations, or cause other problems.

Reply:
The only real concern here is that it can throttle gen 1 Intel CPUs (as all other cores are power/thermal budget based, to the point that AVX512 sometimes lets Zen 4 clock higher if it reduces decode utilization). This has optimal codegen otherwise and makes ARM64 cores very happy.

Would you like me to rewrite this into a manually unrolled version? JIT has gotten very good at these. But if so, I'd like to hear a second opinion on this, if you don't mind. Thanks.

Reply:
That's not the only concern that exists across all potential hardware and targets, there is also implications with regards to AOT, WASM, SVE, potential future hardware that could run using this code, and most importantly the expectation we give to users around our own configuration knobs and that we won't use V256 or V512 if the `DOTNET_PreferredVectorBitWidth` is less than the target.

This is a hard requirement for all SIMD code in the BCL.

Reply:
> But if so, I'd like to hear a second opinion on this, if you don't mind. Thanks.

CC. @stephentoub

Reply:
You are correct, this ignores `DOTNET_PreferredVectorBitWidth` which is not acceptable. I will rewrite it.